compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpCompilerPluginUtil.java

```diff
@@ -184,25 +184,31 @@ private static boolean isServiceType(TypeSymbol returnTypeSymbol) {
 
     private static void validateArrayElementTypeInReturnType(SyntaxNodeAnalysisContext ctx, Node node,
                                                              String typeStringValue, TypeSymbol memberTypeDescriptor,
-                                                             HttpDiagnosticCodes diagnosticCode) {
+                                                             HttpDiagnosticCodes diagnosticCode,
+                                                             boolean isInterceptorType) {
         TypeDescKind kind = memberTypeDescriptor.typeKind();
         if (isAllowedReturnType(kind) || kind == TypeDescKind.RECORD || kind == TypeDescKind.MAP ||
                 kind == TypeDescKind.TABLE) {
             return;
         }
         if (kind == TypeDescKind.INTERSECTION) {
             TypeSymbol typeSymbol = ((IntersectionTypeSymbol) memberTypeDescriptor).effectiveTypeDescriptor();
-            validateArrayElementTypeInReturnType(ctx, node, typeStringValue, typeSymbol, diagnosticCode);
+            validateArrayElementTypeInReturnType(ctx, node, typeStringValue, typeSymbol,
+                    diagnosticCode, isInterceptorType);
         } else if (kind == TypeDescKind.TYPE_REFERENCE) {
             TypeSymbol typeDescriptor = ((TypeReferenceTypeSymbol) memberTypeDescriptor).typeDescriptor();
             TypeDescKind typeDescKind = retrieveEffectiveTypeDesc(typeDescriptor);
-            if (typeDescKind != TypeDescKind.RECORD && typeDescKind != TypeDescKind.MAP &&
-                    typeDescKind != TypeDescKind.TABLE) {
-                reportInvalidReturnType(ctx, node, typeStringValue, diagnosticCode);
+            if (typeDescKind == TypeDescKind.OBJECT) {
+                if (!isHttpModuleType(RESPONSE_OBJ_NAME, typeDescriptor)) {
```

Comment:
We don't allow array of `http:Response`. IIRC incase of `Object` type, we only allow `http:Response`. So in this case we can simply call `reportInvalidReturnType` for `Object` types.
@chamil321 can you also verify?

Reply:
Yes, we only return the `http:Response` as an object from the resource.

Reply:
Updated